### PR TITLE
[WIP] Make the Bluespace Harpoon Require Charges/Battery to Use!

### PR DESCRIPTION
**Pull request by Rykka-Stormheart**
This is not done yet, currently WIP as I'm too tired to keep coherently working. Opening this as a draft so people can review/look at it/give feedback.

TL;DR BS Harpoon is really OP with the literally infinite uses as a result of it being able to be used without a power source required.

This finally wrestles control of this eldritch thing and forces it to use power like any other normal energy-using piece of technology.

IMO, BS Harpoons are still really broken and shouldn't be given to RD's right off roundstart, but if I can get this to work, it'll fix most of the issues I have with it being literally free uses.

### CODE REVIEW

**Comment by Heroman3003** [comment]
Bigger issue with harpoon is not its charge but the way it can straight up invalidate map design in places. Having it straight up not do anything on non-station levels would be enough of a nerf, imo.

**Comment by Rykka-Stormheart** [comment]
> Bigger issue with harpoon is not its charge but the way it can straight up invalidate map design in places. Having it straight up not do anything on non-station levels would be enough of a nerf, imo.

It has neat potential on some maps but on others it breaks everything. Perhaps a flag to prevent BS Harpooning in x z-level specifically would work.

**Comment by ShudderEve** [comment]
This is something that'll punish RD players more than explorers, and I disagree with it.

RDs never give their bluespace harpoons out to exploration, but they get it via diving through trash piles in maintenance.

Removing the bluespace harpoon from trash pile loot should be a fix enough. I don't see a reason why we should nerf RD-only gear because explorers get their hands on it from maintenance. Especially whilst other heads of staff also get equipment that are just as strong if not better at round-start as well.

But if the issue persists and if removing it from trash piles is not a fix enough, removing its use whilst off-site sounds like a suitable fix.

**Comment by lm40** [comment]
tbh i dunno why rd even has a bluespace harpoon

**Comment by cadyn** [comment]
> tbh i dunno why rd even has a bluespace harpoon

I think it's just part of the design of having each head of staff having special items relevant to their department. CMO has medigun, hypospray, Combat defib. HoS has special armored clothes, expandable riot shield, and the cell loaded revolver. CE probably has stuff but I've never played them. And finally RD gets a special AMI hardsuit, some cool looking welding goggles, and the bluespace harpoon.

**Comment by lm40** [comment]
but bluespace harpoon is an outlier in that it isn't really tied to their job. cmo's shit clearly heals, hos' shit clearly securities, ce's shit engineers, hardsuit anomalies and i guess welding goggles are for robotics or something but bluespace harpoon is ????

**Comment by cadyn** [comment]
> but bluespace harpoon is an outlier in that it isn't really tied to their job. cmo's shit clearly heals, hos' shit clearly securities, ce's shit engineers, hardsuit anomalies and i guess welding goggles are for robotics or something but bluespace harpoon is ????

I guess, I still feel like the RD should have (maybe a little biased because I have an RD character), because I know at least I got some good use out of it when I was trying to make a janus mech, was invaluable for being able to enter the burn chamber without venting it for the various PA things that needed specific gases.

**Comment by ShudderEve** [comment]
> but bluespace harpoon is an outlier in that it isn't really tied to their job. cmo's shit clearly heals, hos' shit clearly securities, ce's shit engineers, hardsuit anomalies and i guess welding goggles are for robotics or something but bluespace harpoon is ????

It's baseline ss13 that each head of staff gets special equipment that are not RIGs, which also happen to be objective items for traitors to steal. Whereas on antag servers, RDs usually have a reactive teleport armor which randomly shuffles the wearer around upon being hit, the evasive tool falls short on Virgo due to there being no threat to take damage ever.

The Bluespace Harpoon is an **invaluable** tool for an RD, and is very much related to their job. An RD's primary job is crisis management. Toxins can blow up, xenobio can have an outbreak, xenoarcheology can unearth an anomaly that calls you gay and EMPs the entire z-level. These have all happened quite often, and in these scenarios, a bs harpoon can just yoink anyone in harm's reach from behind doors, and allow medical to have access to them much faster. From my own personal experience, I occasionally had to rely on the harpoon whilst I was teaching others how to do xenobiology. Apart from that, it acts as logistics, and I've not really ever seen it be used to break and enter unlawfully.

I hope we can keep the bs harpoon in the RD's starting equipment, as it is the most useful and thematic tool they have.

**Comment by lm40** [comment]
reactive teleport armor was along the same lines in my opinion in that it didn't really fit. it was just like, hnng, bluespace and teleportation are science-y; but while armor could be useful it didn't tie naturally to the rd's function.

same thing with the harpoon, i have no doubt the harpoon is /useful/. but rd's job is not really the save people that is more of a medical/maybe engineering or security thing.

**Comment by ShudderEve** [comment]
> same thing with the harpoon, i have no doubt the harpoon is /useful/. but rd's job is not really to save people that is more of a medical/maybe engineering or security thing.

It's not that you're saving/aiding people. You're just locating them somewhere where medical is not endangered whilst trying to get to them. It's just the 1st step of the act. Because since we've no antags- if a scientist dies, there's a 90% chance that they have died in a location where they cannot be accessed easily. Just like xenobio during an outbreak, or the outpost with phoron at a temperature of 3000 Kelvins inside as a result of a canister bursting from over-pressurization. The job of an RD is to instruct scientists, help them out, protect the crew from whatever danger that might result from their department, and to manage crises. 

And seeing how every head of staff gets their own piece of equipment- I cannot think of anything better than a very limited teleport at a cooldown of 30 sec - 1 min, that is also very unreliable, as it has a chance of malfunctioning with base components.

I stand by my suggestion to remove bluespace harpoons from maint loot. And a previous conversation in Discord has revealed that whilst the code intends on only 1 bluespace harpoon being able to be looted from trash piles each round- That piece of code is broken, allowing for a much more wider access to it. And this discussion has been generated as part of exploration getting their hands on the bluespace harpoon.

tl;dr: bluespace harpoon is very useful and balanced. It should remain as RD equipment, and just be modified so that it doesn't appear in maintenance as loot, and its use in exploration should be discouraged.

**Comment by lm40** [comment]
yeah and i personally think it's more engaging to have medical/engineering/security whatever have to work together and figure out a way to tactically extract someone instead of just, zap, problem solved. i don't really think harpoon is /harmful/ BECAUSE this isn't an action server, i just think it's a bad thematic fit for the rd to get as head of staff item. i mean it probably shouldn't be removed without replacing it with something because just the hardsuit would be kinda underwhelming when, for instance, hos gets a suit, a gun, and a shield.

any department could make use of a bluespace harpoon to make their job easier, i promise you. that doesn't mean every department should get one. although if it is balanced i am kinda curious why you would want to remove it from maintenance loot piles.

**Comment by ShudderEve** [comment]
> yeah and i personally think it's more engaging to have medical/engineering/security whatever have to work together and figure out a way to tactically extract someone instead of just, zap, problem solved.

It's not a zap, problem solved tool. It's a zap, dead person out tool. So that medical doesn't have to sit and wait outside, the dead person doesn't have to sit in dchat and can be quickly revived, or they don't have to wait for their equipment, and security can deal with whatever threat there is, followed by engineering fixing up the area once more.

There's nothing you can fix with the bluespace harpoon, it's literally just a fun piece of equipment that can be used to yoink corpses.

 
> any department could make use of a bluespace harpoon to make their job easier, i promise you. that doesn't mean every department should get one. although if it is balanced i am kinda curious why you would want to remove it from maintenance loot piles.

Because the main concern I've heard from Heroman and those alike was the fact that exploration acquires a bluespace harpoon to use it off-station.  He originally believed that it was RDs giving their harpoon out, whereas they were just acquiring it through maint. I just do not want the harpoon gone from the RD's kit as per that.

**Comment by lm40** [comment]
you can fix the problem of the dead person being trapped somewhere difficult to reach

anyway i am not really arguing the balance of it or the utility of it or whatever. it would be super unbalanced if we had antagonists but we don't so that is largely a *shrug to me. i suppose it was an attempt to find something /versatile/ since the rd has so many very different departments under them that it would be hard to make stuff for each one. but i think it is too versatile, because most head of staff stuff is basically 'stuff my department has, but better' and the harpoon is really completely different in what it allows you to do.

**Comment by ShudderEve** [comment]
It's very versatile indeed, but it also can't entirely be relied on as a caveat, with a chance of malfunctioning, as well as straight up preventing you from using bluespace bags. If you can think of another item that'd suit the RD better, it's a topic very much open for suggestions, but I think that it's had no harm as is. And it being versatile is not even an issue unless the RD player uses it to do the jobs of others, such is harpooning into mining to get equipment, go mine, then harpoon back along with the crate to get ores themselves. But such behavior is prevented by the rules. Thus, it really doesn't do much except allowing for interesting, fun experiments such as Morphium as Cadyn has suggested.  I still think it's the perfect tool for an RD to have, as well as being significantly less powerful than the CD's roundstart personal translocator, which has no on-screen teleportation limitation, and can instantly teleport anyone to any beacon with a click long as they're not on anything but help intent.

**Comment by lm40** [comment]
tell me sincerely how many times you have used an unupgraded bluespace harpoon

wrt translocator, translocator malfunctions, requires you to be able to physically plant the beacon wherever you're trying to go, so 100% useless on the fly, and it instateleporting other people was supposed to be nerfed

**Comment by ShudderEve** [comment]
> tell me sincerely how many times you have used an unupgraded bluespace harpoon

Never, really. But even the upgraded harpoon is open to malfunction even though the chances are significantly lower.

> wrt translocator, translocator malfunctions, requires you to be able to physically plant the beacon wherever you're trying to go, so 100% useless on the fly, and it instateleporting other people was supposed to be nerfed

A translocator malfunction will only translocate you to a beacon different than the one you've chosen, which only makes you wait for the cooldown. But after you plant your usual set of beacons, you can just hop anywhere on the station on the fly. The only advantage of the BSH is the fact that it can receive things, and help you break-and-enter. But I'd like to think that the latter is enforced by the admins, as breaking into areas as a head of staff to steal departmental gear, or to do another's job is a bannable offense. Spare those two, the CD's translocator is two hundred times better than the bluespace harpoon, in the case of just teleporting around. But to help the department, the bluespace harpoon is preferable.

And- not to mention the powerhouse of teleportation the hand tele is, given that any head of staff has access to it. And- there's two of them on station at the start of any round, in teleporter rooms.

Except the bluespace harpoon, the RD has only the AMI, which is an objectively inferior version of the CE's RIG, and almost even the miner RIG, except the fact that it renders the wearer immune to radiation, with much less overall protection. I think the RD deserves something as versatile. But if there's something that fits the theme better, the topic is very much open for ideas for a replacement item.

**Comment by lm40** [comment]
is it? i may be sadly mistaken but i'm pretty sure it goes to 0 and can actually go negative if you have /ultra high tier/ parts.

give rd hand teleporter instead of the harpoon then, i am totally okay with that if it is such a powerhouse.

**Comment by Wickedtemp** [comment]
tbh same. I don't actually..understand why the RD has it in the first place. Most of the time I've just seen it used to trespass/annoy people. That's probably just because I don't play Science and rarely see it used at all outside of those situations.

For the whole 'its useful for crisis stuff, like getting people out of dangerous areas' ... okay so why does the RD have it instead of engineering/medical. It's engineering's job to make unsafe places safe. It's medical's job to ensure people injured or killed from those unsafe places are all patched up. This tool would make more sense if given to either of those departments.

**Comment by ShudderEve** [comment]
> is it? i may be sadly mistaken but i'm pretty sure it goes to 0 and can actually go negative if you have /ultra high tier/ parts.

Kind of unsure-

> anyway my point remains about the harpoon's unthematic-ness, i already said it's probably not a good idea to remove unless it gets replaced with something. 

I mean, yes. It's not incredibly thematic, except the whole concept around an RD being teleport-ey and bluespace-y. And it doesn't really get used quite often, but neither does any Head of Staff exclusive gear. If there was an alternative solution to it, I think it'd be quite constructive. 

> It's medical's job to ensure people injured or killed from those unsafe places are all patched up. This tool would make more sense if given to either of those departments.

The RD's job includes crisis management for any outbreak or emergency that occurs within their department. Whilst this doesn't mean that they'll carry around medicine and treat their crew, it does mean that they are the first to arrive on the area of emergency, and to yoink the (likely) dead crewmember from danger. The rest is medical + engineering. And, speaking of which, I think that the RD is given minimal gear already.

The HoS gets their NSFW gun, which is incredibly versatile, and has (idk how many available at round start) 8 cells that it can use, each with a different function.

The CE gets **ridiculous** gear, a toolbelt full of advanced tools, an RCD + compressed matter as ammo, a RIG which is objectively superior to the AMI that an RD gets.

The CMO gets a combat defib, which cannot be acquired by any other means than the locker and is objectively superior than their EMTs' compact defib, along with the advanced cell-loaded medigun + cells for it, the prior of which cannot be obtained by any other means either, and is objectively superior to any that can be printed from RnD.

The CD gets a round-start personal translocator barring their suit etc, but I'd not put them in the same category as other heads of staff, given their superiority.

The RD gets their AMI, with a jetpack, pAI module and an Alden-Saraspova counter in it, as well as the Bluespace Harpoon.

I don't see it fit to just remove the harpoon without replacing it with anything, or just handing it to any other head of staff. The only suggestion that I came up with is a round-start Bag of Holding, one that does not fuck up teleportation like a regular one does. If you've any yourselves, this might be an opportunity to make the RD role a little more interesting.

**Comment by lm40** [comment]
> I mean, yes. It's not incredibly thematic, except the whole concept around an RD being teleport-ey and bluespace-y. And it doesn't really get used quite often, but neither does any Head of Staff exclusive gear. If there was an alternative solution to it, I think it'd be quite constructive.

RD is not really teleport-ey tho, telescience doesn't exist, there is nothing in science that is specifically bluespace related except certain R&D items. But saying 'because certain things that can be printed in research relate to this, this is a theme of the rds' would also make the rd an engineer, a doctor, etc. etc. Since all that shit can be made in R&D too.

> > It's medical's job to ensure people injured or killed from those unsafe places are all patched up. This tool would make more sense if given to either of those departments.
> 
> The RD's job includes crisis management for any outbreak or emergency that occurs within their department. Whilst this doesn't mean that they'll carry around medicine and treat their crew, it does mean that they are the first to arrive on the area of emergency, and to yoink the (likely) dead crewmember from danger. The rest is medical + engineering. And, speaking of which, I think that the RD is given minimal gear already.

honestly this is the fundamental disagreement. i really think you're only responsible for crisis management insofar as the crisis relates to your NORMAL departmental duties. for instance, medical is responsible for a virus outbreak, they are not responsible for cleaning up after a chemistry explosion, engineering is responsible for a tesloose structural damage, they are not responsible for treating their own radiation injuries from poorly set up supermatter, etc.

by and large i think most circumstances the rd would use the bluespace harpoon are circumstances where it is really not their mess to clean up. e.g. exploded toxins is engineering for the structure/atmosphere, medical for the wounded.

**Comment by Very-Soft** [comment]
If you give the BSH charges/battery, then take away its chance to fail. As is it can already be dangerous to use, and it would be silly to leave its existing RNG punishment in if we're limiting how much it can be used.

**Comment by nerdass** [comment]
make it not work on non station z-levels instead because right now it's basically the "lol fuck you if you want to design a POI that has windows" gun.

**Comment by Rykka-Stormheart** [comment]
I think I'll be closing this in favor of simply making it not work on non-station levels. I feel that's the most agreeable solution, as well as removing the chance for it to be in random maint loot (so that only RD has it).
